xbmc/guilib/GUITextLayout.cpp

```diff
@@ -540,75 +543,103 @@ void CGUITextLayout::WrapText(const vecText &text, float maxWidth)
   if (!m_font)
     return;
 
-  int nMaxLines = (m_maxHeight > 0 && m_font->GetLineHeight() > 0)?(int)ceilf(m_maxHeight / m_font->GetLineHeight()):-1;
-
   m_lines.clear();
 
+  if (maxWidth < 0)
+  {
+    CLog::LogF(LOGWARNING, "Cannot wrap the text due to invalid max width value.");
+    return;
+  }
+
   std::vector<CGUIString> lines;
   LineBreakText(text, lines);
 
-  for (unsigned int i = 0; i < lines.size(); i++)
+  if (maxWidth == 0) // Unlimited max width
   {
-    const CGUIString &line = lines[i];
-    vecText::const_iterator lastSpace = line.m_text.begin();
+    m_lines = lines;
+    return;
+  }
```

Comment:
Change order like this to avoid a copy for unlimited width

  if (maxWidth == 0) // Unlimited max width
  {
    LineBreakText(text, m_lines);
    return;
  }

  std::vector<CGUIString> lines;
  LineBreakText(text, lines);

Reply:
yep done